program-runtime/src/loaded_programs.rs

```diff
@@ -712,7 +707,10 @@ impl<FG: ForkGraph> LoadedPrograms<FG> {
             if existing.deployment_slot == entry.deployment_slot
                 && existing.effective_slot == entry.effective_slot
             {
-                if matches!(existing.program, LoadedProgramType::Unloaded(_)) {
+                if matches!(
```

Comment:
instead of this, we should do 

```
match existing.program {
    LoadedProgramType::Unloaded(_) | LoadedProgramType::Builtin(_) => {
        // The unloaded program is getting reloaded
        ...
    }
    LoadedProgramType::FailedVerification(_) => ...,
    LoadedProgramType::Closed,
    | LoadedProgramType::DelayVisibility
    | LoadedProgramType::LegacyV0(_)
    ... => ...
```

so we're forced to explicitly express the logic for each variant, which makes it harder to accidentally introduce bugs if the variants change/when the code is updated.